app/Mage.php

```diff
@@ -975,10 +975,42 @@ public static function printException(Throwable $e, $extra = '')
                 print $extra . "\n\n";
             }
             print get_class($e) . ': ' . $e->getMessage() . "\n\n";
-            print $e->getTraceAsString() . "\n";
             if (PHP_SAPI != 'cli') {
-                print '  thrown in <b>' . $e->getFile() . '</b> on line <b>' . $e->getLine() . '</b>' . "\n";
+                $file = $e->getFile();
+                $line = $e->getLine();
+                if (!empty($_SERVER['MAGE_VSCODE_LINKS']) || !empty($_ENV['MAGE_VSCODE_LINKS'])) {
+                    // vscode doc
+                    print preg_replace_callback('#(\#\d+ )([^(]+)\((\d+)\): #', function ($data) {
+                        return
+                            $data[1] .
+                            '<a href="vscode://file/' . $data[2] . ':' . $data[3] . '">' . $data[2] . '</a>' .
+                            '(' . $data[3] . '): ';
+                    }, $e->getTraceAsString()) . "\n";
+                    print '  thrown in <a href="vscode://file/' . $file . ':' . $data[3] . '"><b>' . $file . '</b></a>';
```

Comment:
PHPStan reports that $data is an undefined variable.

Reply:
yes, I used github web file editor... and it as a bad idea